dipy/tracking/streamline.py

```diff
@@ -802,5 +881,119 @@ def values_from_volume(data, streamlines, affine=None):
         raise ValueError("Data needs to have 3 or 4 dimensions")
 
 
+def gaussian_weights(bundle, n_points=100, return_mahalnobis=False,
+                     stat=np.mean):
+    """
+    Calculate weights for each streamline/node in a bundle, based on a
+    Mahalanobis distance from the core the bundle, at that node (mean, per
+    default).
+
+    Parameters
+    ----------
+    bundle : array or list
+        If this is a list, assume that it is a list of streamline coordinates
+        (each entry is a 2D array, of shape n by 3). If this is an array, this
+        is a resampled version of the streamlines, with equal number of points
+        in each streamline.
+    n_points : int, optional
+        The number of points to resample to. *If the `bundle` is an array, this
+        input is ignored*. Default: 100.
+
+    Returns
+    -------
+    w : array of shape (n_streamlines, n_points)
+        Weights for each node in each streamline, calculated as its relative
+        inverse of the Mahalanobis distance, relative to the distribution of
+        coordinates at that node position across streamlines.
+    """
+    if isinstance(bundle, np.ndarray):
+        # It's an array, go with it:
+        n_points = bundle.shape[1]
+    else:
+        # It's something else, assume that it needs to be resampled:
+        bundle = np.array(set_number_of_points(bundle, n_points))
+    w = np.zeros((bundle.shape[0], n_points))
+
+    # If there's only one fiber here, it gets the entire weighting:
+    if bundle.shape[0] == 1:
+        if return_mahalnobis:
+            return np.array([np.nan])
+        else:
+            return np.array([1])
+
+    for node in range(bundle.shape[1]):
+        # This should come back as a 3D covariance matrix with the spatial
+        # variance covariance of this node across the different streamlines
+        # This is a 3-by-3 array:
+        node_coords = bundle[:, node]
```

Comment:
Sorry - it moved now. This is the line I referred to in my previous comment (https://github.com/nipy/dipy/pull/1695/files#r245489936). I don't think that I would be able to do this with something that isn't an array. But I'd be happy to learn that I am wrong.